server/auctionapp/src/main/java/com/atlantbh/auctionapp/service/ProductService.java

```diff
@@ -35,9 +35,19 @@
         return pagedResult;
     }
 
-    public Page<ProductEntity> getAllProductsFromCategory(Integer pageNumber, Integer pageSize, long[] categoryId, double lowPrice, double highPrice){
+    public Page<ProductEntity> getAllProductsFromCategory(Integer pageNumber, Integer pageSize, long[] categoryId, double lowPrice, double highPrice, String sortBy){
         LocalDateTime time = LocalDateTime.now();
-        Pageable paging = PageRequest.of(pageNumber, pageSize, Sort.by("productName"));
+        Sort sortOrder;
+        if (sortBy.equals(SortBy.START_DATE.getSort())){
+            sortOrder = Sort.by(sortBy).descending();
+        } else if (sortBy.equals(SortBy.END_DATE.getSort()) || sortBy.equals(SortBy.START_PRICE.getSort())) {
+            sortOrder = Sort.by(sortBy);
+        } else if (sortBy.equals(SortBy.HIGH_PRICE.getSort())) {
+            sortOrder = Sort.by(SortBy.START_PRICE.getSort()).descending();
+        } else {
+            sortOrder = Sort.by(sortBy);
```

Comment:
I believe we can make this lot simpler. 
if we have:
```
 Sort sort = Sort.by(sortBy);
 sort.and(Sort.by(getSortDirection(sortBy)));
 Pageable paging = PageRequest.of(pageNumber, pageSize, sort);
 ```
 
here getSortDirection would be: 
```
Direction getSortDirection(String sortBy) {
    if (sortBy.equals(SortBy.END_DATE.getSort() || sortBy.equals(SortBy.START_PRICE.getSort()) ) {
        return Direction.ASC;
    } else {
        return Direction.DESC;
    }
}
```